clients/python/test/e2e/ci/e2e/e2e/postprocess.py

```diff
@@ -219,22 +221,34 @@ def clean_up_jobs(artifacts_dir: Path, retry_minutes: Optional[PositiveInt] = No
         msg = "Cleaning up jobs for user: "
         msg += f"\n{server_config.model_dump_json(indent=1)}"
         typer.echo(msg)
-        for attempt in Retrying(
-            retry=retry_if_exception_type(osparc.ApiException),
-            stop=stop_after_delay(timedelta(minutes=retry_minutes))
-            if retry_minutes
-            else stop_after_attempt(1),
-        ):
-            with attempt:
-                with osparc.ApiClient(config) as api_client:
-                    solvers_api = osparc.SolversApi(api_client)
-                    assert isinstance(
-                        solvers := solvers_api.list_solvers_releases(), list
-                    )
-                    for solver in solvers:
-                        assert isinstance(solver, osparc.Solver)
-                        assert (id_ := solver.id) is not None
-                        assert (version := solver.version) is not None
-                        for job in solvers_api.jobs(id_, version):
-                            assert isinstance(job, osparc.Job)
-                            solvers_api.delete_job(id_, version, job.id)
+        try:
+            for attempt in Retrying(
+                retry=retry_if_exception_type((osparc.ApiException, Urllib3HttpError)),
+                stop=stop_after_delay(timedelta(minutes=retry_minutes))
+                if retry_minutes
+                else stop_after_attempt(1),
+            ):
+                with attempt:
+                    with osparc.ApiClient(config) as api_client:
+                        solvers_api = osparc.SolversApi(api_client)
+                        assert isinstance(
+                            solvers := solvers_api.list_solvers_releases(), list
+                        )
+                        for solver in solvers:
+                            assert isinstance(solver, osparc.Solver)
+                            assert (id_ := solver.id) is not None
+                            assert (version := solver.version) is not None
+                            for job in solvers_api.jobs(id_, version):
+                                assert isinstance(job, osparc.Job)
+                                solvers_api.delete_job(id_, version, job.id)
+        except RetryError as exc:
+            typer.echo(
+                typer.style(
+                    (
+                        "Failed when cleaning jobs when encountering "
+                        f"the following exception:\n{exc.last_attempt.exception()}"
+                    ),
+                    fg=typer.colors.RED,
+                    bold=True,
+                )
+            )
```

Comment:
Just an observation: I think no one is going to look at this if the tests are green, right?

Reply:
Probably not. The point is that this is only used to clean up (meaning delete) jobs after tests and is not a test in itself. This sometimes fails if there is any issue on the server which pollutes the e2e. Hence I do it this way so the tests will not fail just because this fails (there are other tests designed to test deletion of jobs). I print the message so at least we have some log of what's going on.